core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -534,38 +534,34 @@ Handle pointToValue(OpData opData) {
         // the old handle index is no longer there so we have nothing to do. Note that in case of non-ZC removal the
         // loop in remove() will lookup the key again so we will still return the previous value
         if (operation == Operation.REMOVE) {
-            return null; // this is a remove, no need to try again and return doesn't matter
+            return true; // this is a remove, no need to try again and return doesn't matter
         }
 
         // the operation is either NO_OP, PUT, PUT_IF_ABSENT, COMPUTE
         int expectedHandleIdx = opData.handleIndex;
         int foundHandleIdx = getEntryField(opData.entryIndex, OFFSET_HANDLE_INDEX);
 
         if (foundHandleIdx == expectedHandleIdx) {
-            return null; // someone helped
+            return true; // someone helped
         } else if (foundHandleIdx < 0) {
             // the handle was deleted, retry the attach
             opData.prevHandleIndex = -1;
             return pointToValue(opData); // remove completed, try again
         } else if (operation == Operation.PUT_IF_ABSENT) {
-            return handles[foundHandleIdx]; // too late
+            return false;
         } else if (operation == Operation.COMPUTE) {
             Handle h = handles[foundHandleIdx];
-            if (h != null) {
-                boolean succ = h.compute(opData.computer);
-                if (!succ) {
-                    // we tried to perform the compute but the handle was deleted,
-                    // we can get to pointToValue with Operation.COMPUTE only from PIACIP
-                    // retry to make a put and to attach the new handle
-                    opData.prevHandleIndex = foundHandleIdx;
-                    return pointToValue(opData);
-                }
+            boolean succ = h.compute(opData.computer);
+            if (!succ) {
+                // we tried to perform the compute but the handle was deleted,
+                // we can get to pointToValue with Operation.COMPUTE only from PIACIP
+                // retry to make a put and to attach the new handle
+                opData.prevHandleIndex = foundHandleIdx;
+                return pointToValue(opData);
             }
-            return h;
         }
-        // this is a put, try again
-        opData.prevHandleIndex = foundHandleIdx;
-        return pointToValue(opData);
+        // this is a put, try again (restart put)
```

Comment:
Maybe we should remove this comment since this particular `return false` also serves for putIfAbsentComputeIfPresent.

Reply:
I prefer to leave this comment here, because for PIACIP it is also for the case of possible put. May be this comment can be enhanced saying "this is put or putIfAbsentComputeIfPresent, try ..."